### PR TITLE
Improvement(#419): Rename animation to sprite animation

### PR DESCRIPTION
**Pull request by renancaraujo**
# Description

According to what has been said on #419, renaming `Animation` to `SpriteAnimation` to avoid misconceptions involving the frameworks `Animation` class.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

If something is unclear, please submit the PR anyways and ask about what you thought was unclear.

- [x] I have added an entry under `[next]` in `CHANGELOG.md`
- [x] I have formatted my code with `flutter format`
- [x] I have made corresponding changes to the documentation
- [x] The continuous integration (CI) is passing

### CODE REVIEW

File: doc/animation_widget_article/article.md [hunk 1]
```diff
@@ -118,18 +118,18 @@ Let's you have your `build` method in one of your pages; pretty normal Flutter s
...
+Note that it could be any component, however complex, inside your widgets tree. Note also that I have omitted the "magic" of the equation here. How is it that we create a component for an animation? Very basically (more details in the flame tutorial), Flame provides components, one of which is the `SpriteAnimationComponent` that receives a `SpriteAnimation` object describing the animation and does exactly what we want. All components live inside a `Game` instance, that can add custom logic relating to the game loop. For our case, we just want to create a simple, empty game and add a single `AnimationComponent` with a simple `Animation` inside. So Flame provides a helper to do that, the `Flame.util.animationAsWidget` method. It takes the size of the object as a Flame's `Position` instance (a generic class to represent a pair of doubles), and also takes in an `Animation` instance representing our frame list. To use that, let's import both `Flame` and the `Animation` class. However, since Flutter adds it's own animation classes, let's use an alias in order to not mess up the names. Therefore, add these imports to the top of the file:
```
Comment (erickzanardo) [inline doc/animation_widget_article/article.md:121]: There is some Animation and AnimationComponent on this sentences yet.

File: doc/animation_widget_article/article.md [hunk 2]
```diff
@@ -118,18 +118,18 @@ Let's you have your `build` method in one of your pages; pretty normal Flutter s
...
 
 ```dart
-import 'package:flame/animation.dart' as animation; // imports the Animation class under animation.Animation
+import 'sprite_animation.dart'; // imports the SpriteAnimation class
```
Comment (erickzanardo) [inline doc/animation_widget_article/article.md:124]: don't remove the `package:flame/` here

File: doc/examples/aseprite/lib/main.dart [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import 'package:flame/flame.dart';
 import 'package:flame/game.dart';
-import 'package:flame/animation.dart' as flame_animation;
+import 'package:flame/sprite_animation.dart' as flame_animation;
```
Comment (erickzanardo) [inline doc/examples/aseprite/lib/main.dart:3]: No need for the alias anymore

File: doc/examples/render_flip/lib/main.dart [hunk 4]
```diff
@@ -1,4 +1,4 @@
-import 'package:flame/animation.dart' as flame_animation;
+import 'package:flame/sprite_animation.dart' as flame_animation;
```
Comment (erickzanardo) [inline doc/examples/render_flip/lib/main.dart:1]: no need for the alias anymore

File: lib/components/animation_component.dart [hunk 5]
```diff
@@ -1,22 +1,22 @@
 import 'dart:ui';
 
 import 'component.dart';
-import 'package:flame/animation.dart';
+import 'package:flame/sprite_animation.dart';
```
Comment (erickzanardo) [inline lib/components/animation_component.dart:4]: Maybe we should change this import to be relative, so we keep it on the standard of the other imports?

File: lib/util.dart [hunk 6]
```diff
@@ -152,14 +152,14 @@ class Util {
...
   /// Returns a regular Flutter widget representing this animation, rendered with the specified size.
   ///
-  /// This actually creates an [EmbeddedGameWidget] with a [SimpleGame] whose only content is an [AnimationComponent] created from the provided [animation].
+  /// This actually creates an [EmbeddedGameWidget] with a [SimpleGame] whose only content is an [SpriteAnimationComponent] created from the provided [animation].
   /// You can use this implementation as base to easily create your own widgets based on more complex games.
   /// This is intended to be used by non-game apps that want to add a sprite sheet animation.
   ///
-  @Deprecated('Use SpriteAnimation instead')
-  widgets.Widget animationAsWidget(Position size, Animation animation) {
+  @Deprecated('Use SpriteAnimationWidget instead')
```
Comment (erickzanardo) [inline lib/util.dart:159]: Since this PR is targeting v1 already, maybe just remove this deprecated method?

File: lib/widgets/animation_widget.dart [hunk 7]
```diff
@@ -1,17 +1,23 @@
 import 'package:flutter/material.dart' hide Animation;
```
Comment (erickzanardo) [inline lib/widgets/animation_widget.dart:1]: no need for this hide anymore

File: lib/widgets/sprite_widget.dart [hunk 8]
```diff
@@ -1,12 +1,20 @@
...
 import 'package:flutter/widgets.dart';
 import 'package:meta/meta.dart';
 import 'dart:math';
 
 import '../sprite.dart';
 import '../anchor.dart';
 
+/// A [StatefulWidget] that renders a still [Sprite].
+///
+/// To render an animation, use [SpriteAnimationWidget].
```
Comment (erickzanardo) [inline lib/widgets/sprite_widget.dart:12]: Nice 👍

File: doc/animation_widget_article/article.md (hunk 1 above)
Comment (spydon) [inline doc/animation_widget_article/article.md:121]: This text needs some language fixes, but I think that is outside of the scope of this PR.

File: lib/components/animation_component.dart (hunk 5 above)
Comment (renancaraujo) [reply lib/components/animation_component.dart:4]: As I have seen on other components, import of files of the first level of teh package are made this way.

Comment (erickzanardo) [reply lib/components/animation_component.dart:4]: Not really, check Sprite.dart for example, all imports a relative, the imports that are absolute from the package are some old ones, anyway this is just a suggestion, this isn't really related to your PR.

File: doc/animation_widget_article/article.md (hunk 1 above)
Comment (renancaraujo) [reply doc/animation_widget_article/article.md:121]: This whole article is obsolete after this PR.

File: lib/components/animation_component.dart (hunk 5 above)
Comment (renancaraujo) [reply lib/components/animation_component.dart:4]: wilco
